From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage of iio_buffer in 4.4.139
Patch-mainline: Never, kABI fix
References: stable-4.4.139

The recent iio fix
  patches.kernel.org/4.4.139-062-iio-buffer-make-length-types-match-kfifo-type.patch
caused a kABI breakage due to the changes from int to unsigned and
size_t.  For fixing this, revert to int for declaration, and use only
unsigned for keeping alignment.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/linux/iio/buffer.h |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

--- a/include/linux/iio/buffer.h
+++ b/include/linux/iio/buffer.h
@@ -49,8 +49,11 @@ struct iio_buffer_access_funcs {
 	int (*request_update)(struct iio_buffer *buffer);
 
 	int (*set_bytes_per_datum)(struct iio_buffer *buffer, size_t bpd);
+#ifdef __GENKSYMS__
+	int (*set_length)(struct iio_buffer *buffer, int length);
+#else
 	int (*set_length)(struct iio_buffer *buffer, unsigned int length);
-
+#endif
 	void (*release)(struct iio_buffer *buffer);
 
 	unsigned int modes;
@@ -78,8 +81,13 @@ struct iio_buffer_access_funcs {
  * @watermark:		[INTERN] number of datums to wait for poll/read.
  */
 struct iio_buffer {
+#ifdef __GENKSYMS__
+	int					length;
+	int					bytes_per_datum;
+#else
 	unsigned int				length;
-	size_t					bytes_per_datum;
+	unsigned int				bytes_per_datum;
+#endif
 	struct attribute_group			*scan_el_attrs;
 	long					*scan_mask;
 	bool					scan_timestamp;
